core/mybatis-generator-core/src/main/java/org/mybatis/generator/codegen/mybatis3/model/ExampleGenerator.java

```diff
@@ -243,9 +243,14 @@ private InnerClass getCriterionInnerClass() {
         method.addBodyLine("this.condition = condition;"); //$NON-NLS-1$
         method.addBodyLine("this.value = value;"); //$NON-NLS-1$
         method.addBodyLine("this.typeHandler = typeHandler;"); //$NON-NLS-1$
-        method.addBodyLine("if (value instanceof List<?>) {"); //$NON-NLS-1$
+        method.addBodyLine("if (value instanceof Collection<?>) {"); //$NON-NLS-1$
```

Comment:
This should be List - not Collection. The method signature that calls this is List.

Reply:
![image](https://user-images.githubusercontent.com/35439368/124087121-4d28f480-da84-11eb-9cc8-9df6f19e2318.png)
It's an Object，I can use Collection like HashSet as well. By the way, what's more important is that  an empty Collection in parameters occurs exception like "java.lang.reflect.InvocationTargetException"